Analysis/Tasks/PWGHF/taskJpsi.cxx

```diff
@@ -0,0 +1,149 @@
+     {"hdeclengthxy", "2-prong candidates;decay length xy (cm);entries", {HistType::kTH1F, {{200, 0., 2.}}}},
+     {"hd0Prong0", "2-prong candidates;prong 0 DCAxy to prim. vertex (cm);entries", {HistType::kTH1F, {{100, -1., 1.}}}},
+     {"hd0Prong1", "2-prong candidates;prong 1 DCAxy to prim. vertex (cm);entries", {HistType::kTH1F, {{100, -1., 1.}}}},
+     {"hd0d0", "2-prong candidates;product of DCAxy to prim. vertex (cm^{2});entries", {HistType::kTH1F, {{500, -1., 1.}}}},
+     {"hCPA", "2-prong candidates;cosine of pointing angle;entries", {HistType::kTH1F, {{110, -1.1, 1.1}}}},
+     {"hEta", "2-prong candidates;candidate #it{#eta};entries", {HistType::kTH1F, {{100, -2., 2.}}}},
+     {"hImpParErr", "2-prong candidates;impact parameter error (cm);entries", {HistType::kTH1F, {{100, -1., 1.}}}},
+     {"hDecLenErr", "2-prong candidates;decay length error (cm);entries", {HistType::kTH1F, {{100, 0., 1.}}}},
+     {"hDecLenXYErr", "2-prong candidates;decay length xy error (cm);entries", {HistType::kTH1F, {{100, 0., 1.}}}}}};
+
+  Configurable<int> d_selectionFlagJpsi{"d_selectionFlagJpsi", 1, "Selection Flag for Jpsi"};
+  Configurable<double> cutEtaCandMax{"cutEtaCandMax", -1., "max. cand. pseudorapidity"};
+
+  Filter filterSelectCandidates = (aod::hf_selcandidate_jpsi::isSelJpsi >= d_selectionFlagJpsi);
+
+  void process(soa::Filtered<soa::Join<aod::HfCandProng2, aod::HFSelJpsiCandidate>> const& candidates)
+  //  void process(aod::HfCandProng2 const& candidates)
+  {
+    //Printf("Candidates: %d", candidates.size());
+    for (auto& candidate : candidates) {
+      if (cutEtaCandMax >= 0. && std::abs(candidate.eta()) > cutEtaCandMax) {
+        //Printf("Candidate: eta rejection: %g", candidate.eta());
+        continue;
+      }
+
+      registry.get<TH1>(HIST("hmass"))->Fill(InvMassJpsi(candidate));
+      registry.get<TH1>(HIST("hptcand"))->Fill(candidate.pt());
+      registry.get<TH1>(HIST("hptprong0"))->Fill(candidate.ptProng0());
+      registry.get<TH1>(HIST("hptprong1"))->Fill(candidate.ptProng1());
+      registry.get<TH1>(HIST("hdeclength"))->Fill(candidate.decayLength());
+      registry.get<TH1>(HIST("hdeclengthxy"))->Fill(candidate.decayLengthXY());
+      registry.get<TH1>(HIST("hd0Prong0"))->Fill(candidate.impactParameter0());
+      registry.get<TH1>(HIST("hd0Prong1"))->Fill(candidate.impactParameter1());
+      registry.get<TH1>(HIST("hd0d0"))->Fill(candidate.impactParameterProduct());
+      registry.get<TH1>(HIST("hCPA"))->Fill(candidate.cpa());
+      registry.get<TH1>(HIST("hEta"))->Fill(candidate.eta());
+      registry.get<TH1>(HIST("hImpParErr"))->Fill(candidate.errorImpactParameter0());
+      registry.get<TH1>(HIST("hImpParErr"))->Fill(candidate.errorImpactParameter1());
+      registry.get<TH1>(HIST("hDecLenErr"))->Fill(candidate.errorDecayLength());
+      registry.get<TH1>(HIST("hDecLenXYErr"))->Fill(candidate.errorDecayLengthXY());
```

Comment:
Update the filling syntax. See https://github.com/AliceO2Group/AliceO2/pull/5150/